### PR TITLE
Small improvements to new AccumuloStore related code

### PR DESCRIPTION
**Pull request by DomGarguilo**

### CODE REVIEW

**Comment by EdColeman** [comment]
Does this set up a merge conflict with #4126 ?

File: core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public abstract class AbstractFateStore<T> implements FateStore<T> {
   private static final Logger log = LoggerFactory.getLogger(AbstractFateStore.class);
 
   protected final Set<Long> reserved;
-  protected final Map<Long,Long> defered;
+  protected final Map<Long,Long> deferred;
```
Comment (cshannon) [inline core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java:55]: This is refactored from the ZooStore and is actually misspelled in 2.1 and main as well so we should fix it there too. https://github.com/apache/accumulo/blob/d97317945290fcd4e3a4eb8b3cf7e427fcc322bd/core/src/main/java/org/apache/accumulo/core/fate/ZooStore.java#L64

File: core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java [hunk 2]
```diff
@@ -62,16 +62,13 @@ public abstract class AbstractFateStore<T> implements FateStore<T> {
...
     this.reserved = new HashSet<>();
-    this.defered = new HashMap<>();
+    this.deferred = new HashMap<>();
   }
 
   public static byte[] serialize(Object o) {
-    try {
-      ByteArrayOutputStream baos = new ByteArrayOutputStream();
-      ObjectOutputStream oos = new ObjectOutputStream(baos);
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
```
Comment (cshannon) [inline core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java:69]: Same comment, this is also an issue in 2.1 https://github.com/apache/accumulo/blob/d97317945290fcd4e3a4eb8b3cf7e427fcc322bd/core/src/main/java/org/apache/accumulo/core/fate/ZooStore.java#L69

File: core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java [hunk 3]
```diff
@@ -75,8 +74,7 @@ protected List<String> getTransactions() {
     return scanTx(scanner -> {
       scanner.setRange(new Range());
       TxColumnFamily.STATUS_COLUMN.fetch(scanner);
-      return StreamSupport.stream(scanner.spliterator(), false)
-          .map(e -> e.getKey().getRow().toString()).collect(Collectors.toList());
+      return scanner.stream().map(e -> e.getKey().getRow().toString()).collect(Collectors.toList());
```
Comment (cshannon) [inline core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java:77]: Nice catch, for some reason I missed the stream() method on the scanner

File: core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java [hunk 4]
```diff
@@ -188,27 +186,23 @@ public long timeCreated() {
...
-        Optional<Integer> top = findTop();
-
-        if (top.filter(t -> t >= maxRepos).isPresent()) {
-          throw new StackOverflowException("Repo stack size too large");
-        }
+      Optional<Integer> top = findTop();
 
-        FateMutator<T> fateMutator = newMutator(tid);
-        fateMutator.putRepo(top.map(t -> t + 1).orElse(1), repo).mutate();
-      } catch (StackOverflowException soe) {
```
Comment (cshannon) [inline core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java:209]: Another nice catch as it makes no sense to catch and immediately re-throw here

**Review by cshannon — APPROVED** [review]
The changes look good, the only thing is the changes in AbstractFateStore exist in 2.1 and main already in ZooStore as that code was refactored in elasticity so we should probably also make the same changes there. I don't think changing things there first would merge forward easily with the refactor so we could just merge this PR and then create another PR to backport the changes to 2.1 and main and then merge an empty commit forward into elasticity

File: core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java (hunk 1 above)
Comment (DomGarguilo) [reply core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java:55]: Looks like the same thing was fixed in #4126 which is waiting to be merged into 2.1 so I think we can just let it be handled there if it ends up being merged.

Comment (DomGarguilo) [reply core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java:55]: Added #4128 to fix the things mentioned in 2.1

**Pull request merged**